src/mpc-hc/MainFrm.cpp

```diff
@@ -12104,7 +12104,8 @@ void CMainFrame::OpenFile(OpenFileData* pOFD)
                 r.fns.AddTail(fn);
                 CPlaylistItem* m_pli = m_wndPlaylistBar.GetCur();
                 if (!m_pli->m_label.IsEmpty()) {
-                    if (!m_pli->m_bYoutubeDL || fn == m_pli->m_ydlSourceURL) r.title = m_pli->m_label;
+                    if (!m_pli->m_bYoutubeDL && PathUtils::StripPathOrUrl(fn) == m_pli->m_label);

```

Comment:
You can change the first if/else to an if statement.
`if (m_pli->m_bYoutubeDL || PathUtils::StripPathOrUrl(fn) != m_pli->m_label) {`

ReverseFind also needs fixing, as that has a char parameter, and not a string pointer.